lib/Base/System/Session.vala

```diff
@@ -132,6 +132,54 @@ public abstract class Backend.Session : Object {
    */
   internal abstract Post[] load_post_list (Json.Node json);
 
+  /**
+   * Favourites/likes a post.
+   *
+   * Adds the favourite/like/platform-equivalent flag to a post. If the post
+   * is already favourited/liked then this is a noop and no exception will be thrown.
+   *
+   * @param post The post to favourite
+   *
+   * @throws Error Any errors while favouriting, such as unauthorised actions, missing posts, or network issues
+   */
+  public abstract async Backend.Post favourite_post (Backend.Post post) throws Error;
+
+  /**
+   * Unfavourites/unlikes a post.
+   *
+   * Removes the favourite/like/platform-equivalent flag from a post. If the post
+   * is not favourited/liked then this is a noop and no exception will be thrown.
+   *
+   * @param post The post to unfavourite
+   *
+   * @throws Error Any errors while unfavouriting, such as unauthorised actions, missing posts, or network issues
+   */
+   public abstract async Backend.Post unfavourite_post (Backend.Post post) throws Error;
+
+  /**
+   * Reblogs/boosts/retweets a post.
+   *
+   * Reblogs a post to the user's timeline. If the post is already reblogged then this is a noop
+   * and no exception will be thrown.
+   *
+   * @param post The post to reblog
+   *
+   * @throws Error Any errors while reblogging, such as unauthorised actions, missing posts, or network issues
+   */
+   public abstract async Backend.Post reblog_post (Backend.Post post) throws Error;
+
+  /**
+   * Un-reblogs/unboosts/un-retweets a post.
+   *
+   * Removes a reblog from the user's timeline. If the post is not reblogged then this is a noop
+   * and no exception will be thrown.
+   *
+   * @param post The post to un-reblog
+   *
+   * @throws Error Any errors while un-reblogging, such as unauthorised actions, missing posts, or network issues
+   */
+   public abstract async Backend.Post unreblog_post (Backend.Post post) throws Error;
+
```

Comment:
I would say this should go into the `Post` class

Reply:
I did debate that. But all of the other calls were in the session, and the Post object seemed to mainly be a holder of data.

Maybe a mid-ground would be cleaner? Post has a "reblog" and "unreblog" function etc, and that internally calls `this.session.reblog_post(this);`?

Reply:
Yes, we could do it that way.